File: src/main/java/brightspark/runicmagic/enums/CanCastResult.java
package brightspark.runicmagic.enums;

import brightspark.runicmagic.RunicMagic;

public enum CanCastResult
{
    SUCCESS(""),
    NO_SPELL("nospell"),
    COOLDOWN("cooldown"),
    LEVEL("level"),
    RUNES("runes"),
    SPELL_REQ("spellreq");

    private final String failLang;

    CanCastResult(String failLang)
    {
        this.failLang = RunicMagic.MOD_ID + ".message.castfail." + failLang;
    }

    public String getFailLang()
    {
        return failLang;
    }
}

File: src/main/kotlin/brightspark/runicmagic/model/CanCastResult.kt
package brightspark.runicmagic.model

import brightspark.runicmagic.RunicMagic

enum class CanCastResult(failLang: String) {
	SUCCESS("success"),
	NO_SPELL("nospell"),
	COOLDOWN("cooldown"),
	LEVEL("level"),
	RUNES("runes"),
	SPELL_REQ("spellreq");

	val failLang: String = "${RunicMagic.MOD_ID}.message.castfail.$failLang"
}
